File: plugins/shelly/src/main/java/eu/geekhome/shellyplugin/ShellyPlugin.java
package eu.geekhome.shellyplugin;

import com.geekhome.common.extensibility.PluginMetadata;
import com.geekhome.common.localization.Resource;
import org.pf4j.Plugin;
import org.pf4j.PluginManager;
import org.pf4j.PluginState;
import org.pf4j.PluginWrapper;

public class ShellyPlugin extends Plugin implements PluginMetadata {

    public static final String PLUGIN_ID_MQTT = "mqtt";

    public ShellyPlugin(PluginWrapper wrapper) {
        super(wrapper);

        PluginManager manager = wrapper.getPluginManager();

        PluginWrapper mqtt = manager.getPlugin(PLUGIN_ID_MQTT);
        if (mqtt.getPluginState() != PluginState.STARTED) {
            manager.startPlugin(PLUGIN_ID_MQTT);
        }
    }

    @Override
    public void start() {
        System.out.println("Starting SHELLY plugin");
    }

    @Override
    public void stop() {
        System.out.println("Stopping SHELLY plugin");
    }

    @Override
    public Resource getName() {
        return new Resource("Shelly", "Shelly");
    }

    @Override
    public Resource getDescription() {
        return new Resource(
                "Unofficial support for Shelly devices",
                "Nieoficjalne wsparcie dla urządzeń Shelly");
    }
}

File: plugins/shelly/src/main/java/eu/geekhome/shellyplugin/ShellyPlugin.kt
package eu.geekhome.shellyplugin

import eu.geekhome.domain.extensibility.PluginMetadata
import eu.geekhome.domain.hardware.HardwarePlugin
import eu.geekhome.domain.langateway.LanGatewayResolver
import eu.geekhome.data.localization.Resource
import eu.geekhome.domain.hardware.HardwareAdapter
import eu.geekhome.domain.mqtt.MqttBrokerService
import org.pf4j.PluginWrapper
import java.util.ArrayList

class ShellyPlugin(
    wrapper: PluginWrapper,
    private val lanGatewayResolver: LanGatewayResolver,
    private val mqttBroker: MqttBrokerService)
    : HardwarePlugin(wrapper), PluginMetadata{

    override fun start() {
        println("Starting SHELLY plugin")
    }

    override fun stop() {
        println("Stopping SHELLY plugin")
    }

    override fun createAdapters(): List<HardwareAdapter<*>> {
        val result = ArrayList<HardwareAdapter<*>>()
        val adapter = ShellyAdapter(pluginId, mqttBroker, lanGatewayResolver)
        result.add(adapter)
        return result
    }

    override val name: Resource =  R.plugin_name
    override val description: Resource = R.plugin_description
}